tests/util/golinter/setup.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+// TestType is type ID of tests
+type TestType int
+
+// All types of tests to parse.
+const (
+	UnitTest  TestType = iota // UnitTest == 0
+	IntegTest TestType = iota // IntegTest == 1
+	E2eTest   TestType = iota // E2eTest == 2
+	NonTest   TestType = iota // NonTest == 3
+)
+
+
+type pathFilter struct {
+	WPaths map[string]map[string]bool // absolute paths that are whitelisted.
+}
+
+func newPathFilter() pathFilter {
+	p := pathFilter{map[string]map[string]bool{}}
+	p.getWhitelistedPathsMap()
+	return p
+}
+
+// getWhitelistedPathsMap converts whitelistedPaths to a map that maps path to rules
+func (pf *pathFilter) getWhitelistedPathsMap() {
+	for path, rules := range WhitelistPath {
+		pf.WPaths[path] = map[string]bool{}
+		ruleList := strings.Split(rules, ",")
+		for _, rule := range ruleList {
+			pf.WPaths[path][rule] = true
+		}
+	}
+}
+
+// IsTestFile checks path absp and desides whether absp is a test file. It returns true and test type
+// for a test file. If path absp should be skipped, it returns false.
+// If one of the following cases meet, path absp is a valid path to test file.
+// (1) e2e test file
+// .../e2e/.../*_test.go
+// (2) integration test file
+// .../integ/.../*_test.go
+// .../integ/.../*_integ_test.go
+// .../*_integ_test.go
+// (3) unit test file
+// .../*_test.go
+func (pf *pathFilter) IsTestFile(absp string, info os.FileInfo) (bool, TestType, map[string]bool) {
```

Comment:
The name should prob be GetTestType instead?

Also, why it this file named setup.go?

Reply:
I rename this file to test_pattern_matcher.go
IsTestFile() is changed to GetTestType(). Thanks!